app/javascript/components/explore/ExploreDisplayTabs.jsx

```diff
@@ -48,6 +48,34 @@ const tabList = [
   { key: 'images', label: 'Images' }
 ]
 
+/** Determine if currently selected annotation has differential expression outputs available */
+function annotHasDe(exploreInfo, exploreParams) {
+  const flags = getFeatureFlagsWithDefaults()
+  if (!flags?.differential_expression_frontend || !exploreInfo) {
+    return false
+  }
+
+  let annotHasDe = false
+  const annotList = exploreInfo.annotationList
+  let selectedAnnot
+  if (exploreParams?.cluster) {
+    selectedAnnot = exploreParams.annotation
+  } else {
+    selectedAnnot = exploreInfo.annotationList.default_annotation
+  }
+
+  const matchingAnnot = annotList.annotations.find(annot => {
```

Comment:
BLOCKER: this duplicates `getMatchedAnnotation` from `cluster-utils.js`

Reply:
Good find!  Some day, a Copilot-like linter will catch these things.  336d270